server/hotrod/src/main/scala/org/infinispan/server/hotrod/Encoder2x.scala

```diff
@@ -312,6 +312,18 @@ object Encoder2x extends AbstractVersionedEncoder with Constants with Log {
          case s: SizeResponse => writeUnsignedLong(s.size, buf)
          case e: ExecResponse =>
             writeRangedBytes(e.result, buf)
+         case r: IterationStartResponse => writeString(r.iterationId, buf)
+         case r: IterationNextResponse =>
+            writeString(r.iterationId, buf)
```

Comment:
Hmmm, having the iteration id in the response might help with async clients, to make it easier to map requests with responses? We do a similar thing with req/resp messages, the message id we receive from clients is sent back in response. To give an example, in the Netty-based Scala async client I wrote, I use the response's message id to complete the request, see [here](https://github.com/galderz/scala-client/blob/master/src/main/scala/org/infinispan/scala/hotrod/impl/CacheClientHandler.scala#L32)